dispatch/apps/events/facebook.py

```diff
@@ -0,0 +1,92 @@
+import urllib2, requests, re, datetime
+
+from django.conf import settings
+
+from dispatch.tests.apis import FacebookTest
+from dispatch.vendor.apis import Facebook, FacebookAPIError
+
+
+class FacebookEvent(object):
+
+    def __init__(self, facebook_url, api_provider=Facebook):
+
+        self.facebook_url = facebook_url
+        self.event_id = self.get_event_id(facebook_url)
+        self.api_provider = api_provider
+
+    def get_event_id(self, facebook_url):
+
+        # Match numbers that is the event id from the facebook_url and return them
+        m = re.search('.*facebook.com/events/([0-9]+).*', facebook_url)
+
+        if m:
+            return m.group(1)
+        else:
+            raise Exception('URL provided is not a valid facebook event url')
+
+    def get_json(self):
+        """Returns the json for the event linked by the facebook url"""
+
+        # Create instance of facebook
+        fb = self.api_provider()
```

Comment:
Perhaps `fb` should be initialized in the constructor since you're doing the same thing in `get_image`

Something like this:

```py

self.api = api_provider()
self.api.get_access_token({
    'client_id': settings.FACEBOOK_CLIENT_ID,
    'client_secret': settings.FACEBOOK_CLIENT_SECRET,
    'grant_type': 'client_credentials'
})